MDANSE/Src/MDANSE/MolecularDynamics/Trajectory.py

```diff
@@ -50,6 +51,68 @@
 SLICE_ALL = np.s_[:]
 
 
+def trajectory_summary(traj: Trajectory):
+    val = []
+    try:
+        time_axis = traj.time()
+    except Exception:
+        timeline = "No time information!\n"
+    else:
+        if len(time_axis) < 1:
+            timeline = "N/A\n"
+        elif len(time_axis) < 5:
+            timeline = f"{time_axis}\n"
+        else:
+            timeline = f"[{time_axis[0]}, {time_axis[1]}, ..., {time_axis[-1]}]\n"
+
+    val.append("Path:")
+    val.append(f"{traj.filename}\n")
+    val.append("Number of steps:")
+    val.append(f"{len(traj)}\n")
+    val.append("Configuration:")
+    val.append(f"\tIs periodic: {traj.unit_cell(0) is not None}\n")
+    try:
+        val.append(f"First unit cell (nm):\n{traj.unit_cell(0)._unit_cell}\n")
+    except Exception:
+        val.append("No unit cell information\n")
+    val.append("Frame times (1st, 2nd, ..., last) in ps:")
+    val.append(timeline)
+    val.append("Variables:")
+    for k in traj.variables():
+        v = traj.variable(k)
+        try:
+            val.append(f"\t- {k}: {v.shape}")
+        except AttributeError:
+            try:
+                val.append(f"\t- {k}: {v['value'].shape}")
+            except KeyError:
+                continue
+
+    val.append("\nConversion history:")
+    metadata = check_metadata(traj.file)
+    if metadata:
+        for k, v in metadata.items():
+            val.append(f"{k}: {v}")
```

Comment:
I feel like the number of times we do this we want an IOUtils function for this. 

Not for this PR though.